services/utils/__init__.py

```diff
@@ -101,24 +102,68 @@ def __init__(self,
                  pool_min: int = 1,
                  pool_max: int = 10,
                  timeout: int = 60):
-        table = str.maketrans({"'": "\'", "`": r"\`"})
 
         self._dsn = os.environ.get(prefix + "DSN", dsn)
-
-        self.host = os.environ.get(prefix + "HOST", host).translate(table)
-        self.port = int(os.environ.get(prefix + "PORT", port))
-        self.user = os.environ.get(prefix + "USER", user).translate(table)
-        self.password = os.environ.get(
-            prefix + "PSWD", password).translate(table)
-        self.database_name = os.environ.get(
-            prefix + "NAME", database_name).translate(table)
+        # We explicitly check the validity of the DSN string 
+        # to avoid issues caused by bad DSN strings. 
+        if not self._is_valid_dsn(self._dsn):
```

Comment:
if the DSN is set but invalid, wouldn't it be better to just fail loudly?

Reply:
Done.